scripts/generate_manifest.py

```
@@ -0,0 +1,40 @@
+            return f"{part[:4]}-{part[4:6]}-{part[6:]}"
+    return "unknown"
+
+
+def generate_manifest(raw_path, processed_path):
+    raw_path = Path(raw_path)
+    processed_path = Path(processed_path)
+
+    if not processed_path.exists():
+        raise FileNotFoundError(
+            f"Processed file not found at {processed_path}. Run preprocessing first."
+        )
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)